pkg/types/powervs/validation/platform.go

```diff
@@ -6,17 +6,56 @@ import (
 	"github.com/openshift/installer/pkg/types/powervs"
 )
 
+var powerVSRegions = func() map[string]string {
+	regions := make(map[string]string)
+
+	for _, region := range powervs.Regions {
+		regions[region.Name] = regions[region.Description]
+	}
+	return regions
+}()
+
+var regionShortNames = func() []string {
+	keys := make([]string, len(powerVSRegions))
+	i := 0
+	for r := range powerVSRegions {
+		keys[i] = r
+		i++
+	}
+	return keys
+}()
+
```

Comment:
i'd prefer this to be a bit simpler if all you need is a list of the names:

```@@ -6,23 +6,12 @@ import (
        "github.com/openshift/installer/pkg/types/powervs"
 )
 
-var powerVSRegions = func() map[string]string {
-       regions := make(map[string]string)
-
-       for _, region := range powervs.Regions {
-               regions[region.Name] = regions[region.Description]
-       }
-       return regions
-}()
-
 var regionShortNames = func() []string {
-       keys := make([]string, len(powerVSRegions))
-       i := 0
-       for r := range powerVSRegions {
-               keys[i] = r
-               i++
+       var shortnames []string
+       for name := range powervs.Regions {
+               shortnames = append(shortnames, name)
        }
-       return keys
+       return shortnames
 }()
 
 // ValidatePlatform checks that the specified platform is valid.
@@ -31,7 +20,7 @@ func ValidatePlatform(p *powervs.Platform, fldPath *field.Path) field.ErrorList
 
        if p.Region == "" {
                allErrs = append(allErrs, field.Required(fldPath.Child("region"), "region must be specified"))
-       } else if _, ok := powerVSRegions[p.Region]; !ok {
+       } else if _, ok := powervs.Regions[p.Region]; !ok {
                allErrs = append(allErrs, field.NotSupported(fldPath.Child("region"), p.Region, regionShortNames))
        }
```

Reply:
fixed